### PR TITLE
Fixed the installation directory on Linux

### PR DESCRIPTION
**Pull request by Mailaender**
The main underlying problem is that ImHex does not (yet) adhere to https://refspecs.linuxfoundation.org/fhs.shtml and https://specifications.freedesktop.org/basedir-spec so I covered cmake with workarounds. Sorry.

### CODE REVIEW

File: cmake/build_helpers.cmake [hunk 1]
```diff
@@ -188,7 +188,7 @@ macro(createPackage)
     endif()
 
     if (UNIX AND NOT APPLE)
-        install(TARGETS libimhex DESTINATION ${CMAKE_INSTALL_PREFIX})
+        install(TARGETS libimhex DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/imhex)
```
Comment (Tatsh) [inline cmake/build_helpers.cmake:191]: I think this should go in `CMAKE_INSTALL_LIBDIR`. By default that's in search paths.

**Comment by Tatsh** [comment]
Also please see my patch: https://github.com/Tatsh/tatsh-overlay/blob/master/app-editors/imhex/files/imhex-installation.patch

**Review by Tatsh — CHANGES_REQUESTED** [review]

File: cmake/build_helpers.cmake (hunk 1 above)
Comment (Mailaender) [reply cmake/build_helpers.cmake]: Yep, that's way better.

File: cmake/build_helpers.cmake [hunk 2]
```diff
@@ -215,11 +215,19 @@ macro(createPackage)
             )
 
     foreach (plugin IN LISTS PLUGINS)
-        install(FILES "$<TARGET_FILE:${plugin}>" DESTINATION plugins/)
+        if(UNIX AND NOT APPLE)
+            install(FILES "$<TARGET_FILE:${plugin}>" DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/imhex/plugins/)
```
[the remark this answers is not in the document]
Comment (Mailaender) [reply cmake/build_helpers.cmake:228]: Nope, plugins are binary and are not allowed in DATADIR according to FHS.

File: cmake/build_helpers.cmake [hunk 3]
```diff
@@ -240,6 +248,16 @@ macro(createPackage)
...
+        elseif(UNIX AND NOT APPLE)
+            install(TARGETS imhex RUNTIME DESTINATION ${CMAKE_INSTALL_LIBEXECDIR}/imhex)
+
+            file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/wrapper "#!/bin/sh
+SETTINGS_FOLDER=$HOME/.local/share/imhex
+mkdir -p $SETTINGS_FOLDER
+cd $SETTINGS_FOLDER
+${CMAKE_INSTALL_LIBEXECDIR}/imhex/imhex")
+
+            install(FILES ${CMAKE_CURRENT_BINARY_DIR}/wrapper DESTINATION ${CMAKE_INSTALL_BINDIR} RENAME imhex PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
```
[the remark this answers is not in the document]
Comment (Mailaender) [reply cmake/build_helpers.cmake]: Removed it.

**Comment by Mailaender** [comment]
I redid this according to your patch which works really well. I only changed plugin path and lower cased (as that was done to the binary as well recently) the paths. Tried to rebase this, but it seems a lot has changed. I also don't fully understand the changes. Everything seems to be pulled from XDG user directories on Linux which makes no sense. I guess we will have to patch this all over again for the next release. :(

File: cmake/build_helpers.cmake (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mailaender) [reply cmake/build_helpers.cmake]: Though that makes ImHex put its config files at the location where you type the execution command...

**Comment by Mailaender** [comment]
https://build.opensuse.org/request/show/880264

**Pull request closed without merging**